plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiComponentRegistrar.kt

```diff
@@ -14,13 +16,23 @@ import java.io.File
 
 class JvmAbiComponentRegistrar : ComponentRegistrar {
     override fun registerProjectComponents(project: MockProject, configuration: CompilerConfiguration) {
-        @Suppress("NAME_SHADOWING")
-        val configuration = configuration.copy()
-        configuration.get(JvmAbiConfigurationKeys.OUTPUT_DIR)?.let {
-            val dir = File(it)
-            configuration.put(JVMConfigurationKeys.OUTPUT_DIRECTORY, dir)
+        val outputPath = configuration.getNotNull(JvmAbiConfigurationKeys.OUTPUT_PATH)
+        if (configuration.get(JvmAbiConfigurationKeys.LEGACY_ABI_GEN, false)) {
```

Comment:
Do I understand correctly that new scheme would be default one after merging in 1.5.30?
I would prefer to announce it as non-default option in 1.5.30 if it's ok and make default in 1.6 // cc @udalov

Reply:
It seems that the plugin won't make it to 1.5.30, we hope to merge it into 1.6-M1.

I think it's fine to change the behavior right away, since the new scheme is objectively better and that way we'll gather feedback quicker. Also, there's not a lot of users of the plugin, so I don't think it'll break a lot.

Reply:
There are also known issues with the old scheme (e.g., KT-40133 and KT-40340), which are fixed with the new implementation.